### PR TITLE
language/python: allow python@3.7 to be used for virtualenvs

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: Library/Homebrew/exceptions.rb [hunk 1]
```diff
@@ -320,6 +320,18 @@ def message
...
 end
 
+class FormulaUnknownPythonError < RuntimeError
+  def initialize(formula)
+    super <<~EOS
+      The version of Python to use with the virtualenv in the `#{formula.full_name}` formula
+      cannot be guessed automatically because a recognised Python dependency could not be found.
+
+      If you are using a non-standard Python depedency, please add `:using => "python@x.y"` to
+      `virtualenv_install_with_resources` to resolve the issue manually.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/exceptions.rb:330]: ❤️

**Comment by Rylan12** [comment]
Is it worth changing this to accept any python dependency without hardcoding an allowlist?

i.e. check for a `python@` dependency instead of only `python@3.7` or `python@3.8` or `python@3`

It gives us one less thing to change when `python@3.7` is removed or `python@3.9` is added

**Comment by Bo98** [comment]
Yes, it is definitely worth doing that.

I might opt to do that after the brew release today (don't want to set up too many road blocks) unless you all prefer for me to do it now.

**Comment by MikeMcQuaid** [comment]
👍 to holding off for now so we can land this PR and get a release out ASAP.

**Comment by Rylan12** [comment]
👍 totally agree

**Comment by petetnt** [comment]
After this PR `brew install eugenmayer/dockersync/unox` fails for me for some reason.

Python itself seems to work just fine:

```
> python
> Python 3.8.5 (default, Oct  1 2020, 18:45:09)                                                                                                                                                     
```

Same problem if aliasing to python 2. MacOS 10.15.5.

**Pull request merged**